internal/rebooter/reconcile.go

```diff
@@ -0,0 +1,132 @@
+package rebooter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"nebius.ai/slurm-operator/internal/controller/reconciler"
+	"nebius.ai/slurm-operator/internal/controllerconfig"
+)
+
+var (
+	ControllerName  = "rebooter"
+	SlurmNodeDrain  = corev1.NodeConditionType("SlurmNodeDrain")
+	SlurmNodeReboot = corev1.NodeConditionType("SlurmNodeReboot")
```

Comment:
not in this PR, just not to forget
maybe we can have separate file closer to the checks controller to store all the conditions there